src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -0,0 +1,519 @@
+}
+
+void DockerVirtualMachinePrivate::doRestoreSnapshot(const QString &snapshotName,
+        const QObject *context, const Functor<bool> &functor)
+{
+    Q_UNUSED(snapshotName)
+    Q_UNUSED(context)
+    Q_UNUSED(functor)
+    QTC_CHECK(false);
+}
+
+VirtualMachineInfo DockerVirtualMachinePrivate::virtualMachineInfoFromOutput(
+        const QByteArray &output) const
+{
+    VirtualMachineInfo info;
+
+    QJsonDocument document = QJsonDocument::fromJson(output);
+    QTC_ASSERT(!document.isNull(), return info);
+
+    info.memorySizeMb = VirtualMachine::availableMemorySizeMb();
+    info.cpuCount = VirtualMachine::availableCpuCount();
+
+    QJsonObject labels = document.object();
+
+    auto parsePort = [=](VirtualMachinePrivate::ReservedPort which, quint16 *port) {
+        const QString key = labelFor(which);
+        QTC_ASSERT(labels.contains(key), return);
+        *port = labels.value(key).toString().toUShort();
+    };
+
+    parsePort(VirtualMachinePrivate::SshPort, &info.sshPort);
+    parsePort(VirtualMachinePrivate::WwwPort, &info.wwwPort);
+
+    auto parsePath = [=](VirtualMachinePrivate::SharedPath which, QString *path) {
+        const QString key = labelFor(which);
+        QTC_ASSERT(labels.contains(key), return);
+        *path = labels.value(key).toString();
+    };
+
+    parsePath(VirtualMachinePrivate::SharedSrc, &info.sharedSrc);
+    parsePath(VirtualMachinePrivate::SharedSsh, &info.sharedSsh);
+    parsePath(VirtualMachinePrivate::SharedHome, &info.sharedHome);
+    parsePath(VirtualMachinePrivate::SharedConfig, &info.sharedConfig);
+    parsePath(VirtualMachinePrivate::SharedTargets, &info.sharedTargets);
+
+    return info;
+}
+
+QString DockerVirtualMachinePrivate::labelFor(VirtualMachinePrivate::SharedPath which)
+{
+    switch (which) {
+    case SharedSrc:
+        return Constants::BUILD_ENGINE_SHARED_SRC;
+    case SharedHome:
+        return Constants::BUILD_ENGINE_SHARED_HOME;
+    case SharedConfig:
+        return Constants::BUILD_ENGINE_SHARED_CONFIG;
+    case SharedSsh:
+        return Constants::BUILD_ENGINE_SHARED_SSH;
+    case SharedTargets:
+        return Constants::BUILD_ENGINE_SHARED_TARGET;
+    }
+    Q_ASSERT(false);
+    return QString();
+}
+
+QString DockerVirtualMachinePrivate::labelFor(VirtualMachinePrivate::ReservedPort which)
+{
+    switch (which) {
+    case SshPort:
+        return Constants::BUILD_ENGINE_SSH_PORT;
+    case WwwPort:
+        return Constants::BUILD_ENGINE_WWW_PORT;
+    }
+    Q_ASSERT(false);
+    return QString();
+}
+
+QString DockerVirtualMachinePrivate::dockerPath()
+{
+    static QString s_dockerPath;
+    if (s_dockerPath.isNull()) {
+        QStringList searchPaths = QProcessEnvironment::systemEnvironment()
+            .value("PATH").split(QChar(':'));
+        foreach (const QString &searchPath, searchPaths) {
+            QDir dir(searchPath);
+            if (dir.exists(DOCKER)) {
+                s_dockerPath=dir.absoluteFilePath(DOCKER);
```

Comment:
Spaces

Reply:
I was looking for the variable in class declaration... only now I noticed it is local to this function :) omit the `s_` please.

Reply:
fixed